src/GraphQL.MicrosoftDI/ScopedAttribute.cs

```diff
@@ -0,0 +1,21 @@
+using GraphQL.MicrosoftDI;
+using GraphQL.Types;
+
+namespace GraphQL
+{
+    /// <summary>
+    /// Creates a dedicated service scope during the field resolver's execution.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method)]
+    public class ScopedAttribute : GraphQLAttribute
+    {
+        /// <inheritdoc/>
+        public override void Modify(FieldType fieldType, bool isInputType)
+        {
```

Comment:
```suggestion
        {
            Debug.Assert(isInputType == false);
```
?

Reply:
Sure, if you like but it would only verify that the tests were written correctly. It won’t prevent users tagging input fields. But since an input field does not have a field resolver, this attribute will do nothing. (See if condition on next line of code.)

Reply:
I would make condition expliciit: `if (isInputType) return;`